pilot/pkg/config/kube/gateway/inferencepool_collection.go

```diff
@@ -98,140 +98,348 @@ func InferencePoolCollection(
+		finalParents = append(finalParents, parentStatus)
+	}
+
+	return &inferencev1alpha2.InferencePoolStatus{
+		Parents: finalParents,
+	}
+}
+
+// findGatewayParents finds all Gateway parents that reference this InferencePool through HTTPRoutes
+func findGatewayParents(
+	pool *inferencev1alpha2.InferencePool,
+	routeList []*gateway.HTTPRoute,
+) sets.Set[types.NamespacedName] {
+	gatewayParents := sets.New[types.NamespacedName]()
+
+	for _, route := range routeList {
+		// Only process routes that reference our InferencePool
+		if !routeReferencesInferencePool(route, pool) {
+			continue
+		}
+
+		// Check the route's parent status to find accepted gateways
+		for _, parentStatus := range route.Status.Parents {
+			// Only consider parents managed by our supported controllers (from supportedControllers variable)
+			// This filters out parents from other controllers we don't manage
+			if !supportedControllers.Contains(parentStatus.ControllerName) {
+				continue
 			}
-			if pool.Spec.ExtensionRef.PortNumber != nil {
-				extRef.port = int32(*pool.Spec.ExtensionRef.PortNumber)
-			} else {
-				extRef.port = 9002 // Default port for the inference extension
+
+			// Get the gateway namespace (default to route namespace if not specified)
+			gatewayNamespace := route.Namespace
+			if parentStatus.ParentRef.Namespace != nil && *parentStatus.ParentRef.Namespace != "" {
+				gatewayNamespace = string(*parentStatus.ParentRef.Namespace)
 			}
 
-			svcName, err := InferencePoolServiceName(pool.Name)
-			if err != nil {
-				log.Errorf("failed to generate service name for InferencePool %s: %v", pool.Name, err)
-				return nil, nil
+			gatewayParents.Insert(types.NamespacedName{
+				Name:      string(parentStatus.ParentRef.Name),
+				Namespace: gatewayNamespace,
+			})
+		}
+	}
+
+	return gatewayParents
+}
+
+// routeReferencesInferencePool checks if an HTTPRoute references the given InferencePool
+func routeReferencesInferencePool(route *gateway.HTTPRoute, pool *inferencev1alpha2.InferencePool) bool {
+	for _, rule := range route.Spec.Rules {
+		for _, backendRef := range rule.BackendRefs {
+			if !isInferencePoolBackendRef(backendRef.BackendRef) {
+				continue
 			}
 
-			shadowSvcInfo := shadowServiceInfo{
-				key: types.NamespacedName{
-					Name:      svcName,
-					Namespace: pool.GetNamespace(),
-				},
-				selector:   make(map[string]string, len(pool.Spec.Selector)),
-				poolName:   pool.GetName(),
-				targetPort: pool.Spec.TargetPortNumber,
-				poolUID:    pool.GetUID(),
+			// Check if this backend ref points to our InferencePool
+			if string(backendRef.BackendRef.Name) != pool.ObjectMeta.Name {
+				continue
 			}
 
-			for k, v := range pool.Spec.Selector {
-				shadowSvcInfo.selector[string(k)] = string(v)
+			// Check namespace match
+			backendRefNamespace := route.Namespace
+			if backendRef.BackendRef.Namespace != nil && *backendRef.BackendRef.Namespace != "" {
+				backendRefNamespace = string(*backendRef.BackendRef.Namespace)
```

Comment:
```suggestion
			// Check namespace match
			gatewayNamespace := ptr.NonEmptyOrDefault(parentStatus.ParentRef.Namespace,
				ptr.Of(gatewayv1.Namespace(route.Namespace)))
```

@keithmattix Technically we can do this too.. I'm just not convinced it's an improvement with the type casting and extra ptr call. wdyt?